lib/code_corps/accounts/accounts.ex

```diff
@@ -6,20 +6,41 @@ defmodule CodeCorps.Accounts do
   """
 
   alias CodeCorps.{
+    Accounts,
     Accounts.Changesets,
+    Analytics,
     Comment,
     GitHub.Adapters,
     GithubAppInstallation,
     GithubUser,
     Processor,
     Task,
     User,
+    UserInvite,
     Repo
   }
   alias Ecto.{Changeset, Multi}
 
   import Ecto.Query
 
+  @doc ~S"""
+  Creates a `CodeCorps.User` account when such a request is made from the client
+  application.
+
+  If an `invite_id` attribute is provided in the request, treats the process as
+  claiming the specified invite.
+  """
+  @spec create(map) :: {:ok, User.t} | {:error, Changeset.t} | {:error, :invite_not_found}
+  def create(%{"invite_id" => _} = params) do
+    with {:ok, %User{} = user, %UserInvite{} = user_invite} <- params |> Accounts.UserInvites.claim_invite() do
+      user.id |> Analytics.SegmentTracker.track("Claimed User Invite", user_invite)
+      {:ok, user}
+    end
+  end
+  def create(%{} = params) do
+    %User{} |> User.registration_changeset(params) |> Repo.insert()
+  end
```

Comment:
We needed to extract creation into a context. I didn't have a better idea other than this multiple clause approach to differentiate between plain account creation and an invite claim.